src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -364,15 +364,39 @@ export class RunParametersAction extends TooltipFromLabelAction {
 	public async openParameterizedNotebook(uri: URI): Promise<void> {
 		const editor = this._notebookService.findNotebookEditor(uri);
 		let modelContents = JSON.stringify(editor.model.toJSON());
-		let basename = path.basename(uri.fsPath);
-		let untitledUri = uri.with({ authority: '', scheme: 'untitled', path: basename });
+		let listNotebookEditors: INotebookEditor[] = this._notebookService.listNotebookEditors();
+		let untitledUriPath = getUntitledUriPath(path.basename(uri.fsPath), listNotebookEditors);
+		let untitledUri = uri.with({ authority: '', scheme: 'untitled', path: untitledUriPath });
 		this._notebookService.openNotebook(untitledUri, {
 			initialContent: modelContents,
 			preserveFocus: true
 		});
 	}
 }
 
+/**
+ * Will iterate the title of the parameterized notebook since the original notebook is still open
+ * @param originalTitle is the title of the original notebook that we run parameterized action from
+ * @param listNotebookEditors is the list of notebooks editors currently open
+ * @returns the title of the parameterized notebook
+ */
+export function getUntitledUriPath(originalTitle: string, listNotebookEditors: INotebookEditor[]): string {
```

Comment:
This seems like a more generally useful thing - not really specific to parameterized notebooks (we already do something similar in the extension).

I'd suggest making this a common function in the service code so it's more easily discoverable